### PR TITLE
Issue 83

### LINKED ISSUE #83: Pre-Schematron Structured FedRAMP Reviewer Validations Rule Data
As the FedRAMP PMO and ASAP P3 developer, in order to better understand how the FedRAMP Validations project correctly implements each business rule reviewers manually validate in my ongoing processes, I want to see a more structured list of these rules and how they map to the validation developers' code.

**Context:**

Onboarded 10x developers and FedRAMP stakeholders have discussed a need for more structured information than what was provided previously to the 10x ASAP P2 Team last year in the form of the [Agency Review Report Template](https://docs.google.com/spreadsheets/d/1Rcw7KvUudIPMSMmo0lbhXE3QgqqaS76S0BmFC8ThrXA/edit#gid=177951143). The current developers would like a more intentional, structured precursor document with a listing of rules that can be reviewed and modified over time. Once this precursor document is established, the developers would like to investigate more directly mapping it to the code outputs (in later development efforts, perhaps after Sprint 2), perhaps using it to programmatically generate tests and/or the validation code itself. 

Nice to have: A rendition of this structured precursor document into HTML or an alternative presentation format with the FART and JAB team would be ideal.

**Acceptance:**

- [ ] Documentation of approach and a minimal example of one structured business rule with required data
- [ ] A structured presentation of our understanding on FedRAMP business rules
- [ ] Document our understanding of what triggers changes in FedRAMP business rules (i.e. what drives FedRAMP rule changes): 
  - NIST baseline changes (OMB mandates this adoption)
  - OSCAL releases? Is this just a cascade effect to NIST change, specimen SSP
  - Internal process improvement

**Dependencies:**
- [ ] A working agreement with FedRAMP FART and JAB contacts to align on this structured data

### PR DESCRIPTION
**Pull request by GaryGapinski**
This PR establishes a set of "business rules" which are associated with Schematron validation. An associated XML Schema document is provided.

An HTML report is included.

Closes #83.

### CODE REVIEW

File: src/validations/rules/rules.xsl [hunk 1]
```diff
@@ -1,7 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <xsl:stylesheet
-    exclude-result-prefixes="xs math oscal fv fn sch doc x"
+    exclude-result-prefixes="xs math oscal fv fn sch doc msg x"
     version="3.0"
```
Comment (danielnaab) [inline src/validations/rules/rules.xsl:4]: How hard would it be to add this to the build, so we can easily view via the Federalist deploy?

Comment (danielnaab) [reply src/validations/rules/rules.xsl:4]: I mean add rules.xsl as the rendered HTML.

Comment (GaryGapinski) [reply src/validations/rules/rules.xsl:4]: We can render HTML (using `rules.xsl` with `rules.xml`+`ssp.sch` as inputs) and produce it as a Federalist deployment artifact. I'd be happy to do so if you can lend a bit of your expertise. All that would be required is a single Saxon-mediated XSL transform.

File: src/validations/rules/rules.xml [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+        doc:guide-reference="DRAFT Guide to OSCAL-based FedRAMP System Security Plans §6"
+        doc:template-reference="System Security Plan Template §15">
+        <statement>FedRAMP SSP policy and procedure documents must have unique per-control-family associations.</statement>
+    </rule>
+    <rule
+        assertions="has-user-guide"
+        doc:attachment="§15 Attachment 2"
+        doc:checklist-reference="Section B Check 3.2"
+        doc:guide-reference="DRAFT Guide to OSCAL-based FedRAMP System Security Plans §6"
+        doc:template-reference="System Security Plan Template §15">
```
[the remark this answers is not in the document]
Comment (GaryGapinski) [reply src/validations/rules/rules.xml:43]: @danielnaab : did you mean remove the attribute (e.g., `doc:attachment="§15 Attachment 2"`) or the related text in the assertion message (e.g., `[Section B Check 3.8]`)?

I removed the latter in #193.

File: src/validations/rules/rules.xsl (hunk 1 above)
Comment (GaryGapinski) [reply src/validations/rules/rules.xsl:4]: Or did you mean add `rules.xsl `to the UI as an available transform? A transform to HTML or XML or JSON?

File: src/validations/rules/rules.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielnaab) [reply src/validations/rules/rules.xml:43]: I meant both, but would defer to you on what makes the most sense.

File: src/validations/rules/rules.xsl (hunk 1 above)
Comment (danielnaab) [reply src/validations/rules/rules.xsl:4]: Sorry for the delay in responding - I meant create the target HTML file. I could coordinate with you on how to do that, and I don't think it needs to be done for this PR.

**Review by danielnaab — APPROVED** [review]
Looks great - however you want to handle duplication works for me

**Pull request merged**